jwst/barshadow/bar_shadow.py

```diff
@@ -0,0 +1,440 @@
+from __future__ import division
+
+#
+#  Module for calculating bar shadow correction for science data sets
+#
+
+import numpy as np
+import logging
+from .. import datamodels
+from jwst.assign_wcs import nirspec
+from gwcs import wcstools
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+slit_spacing = 202.0
+slit_height = 175.0
+SLITRATIO = slit_spacing/slit_height     # Approximate value until we have something better
+
+def do_correction(input_model, barshadow_model):
+    """
+    Short Summary
+    -------------
+    Execute all tasks for Bar Shadow Correction
+
+    Parameters
+    ----------
+    input_model: data model object
+        science data to be corrected
+
+    barshadow_model: barshadow model object
+        bar shadow correction data
+
+    Returns
+    -------
+    output_model: data model object
+        Science data with bar shadow extensions added
+
+    """
+
+#
+# Input is a MultiSlitModel science data model 
+# A MultislitModel has a member .slits that behaves like
+# a list of Slits, each of which has several data arrays and
+# keyword metadata items associated 
+#
+# At this point we're going to have to assume that a Slit is composed
+# of a set of slit[].nshutters in a vertical line
+#
+# Reference file information is a 1x1 ref file and a 1x3 ref file
+# Both the 1x1 and 1x3 ref files are 1001 pixel high and go from
+# -1 to 1 in their Y value WCS.  
+# 
+# 
+    exp_type = input_model.meta.exposure.type
+    log.info(exp_type)
+    #
```

Comment:
Maybe add a check at this point that skips (returns) right away if exp_type is not NRS_MSASPEC.

Reply:
Or alternatively just move the first 3 lines of code below to be inside of the "if exp_type" block, so that we don't waste time doing things that aren't necessary.